flink/v1.18/flink/src/main/java/org/apache/iceberg/flink/source/reader/IcebergSourceSplitReader.java

```diff
@@ -113,7 +139,13 @@ public void handleSplitsChanges(SplitsChange<IcebergSourceSplit> splitsChange) {
   }
 
   @Override
-  public void wakeUp() {}
+  public void wakeUp() {
+    LOG.info("WakeUp called");
+    pausedSplits.clear();
```

Comment:
The `fetch` method to be unblocked upon `wakeUp`. but is it correct to clear the `pausedSplits` upon `wakeUp`?

Reply:
You are right. The wakeUp is called whenever there is a new split, so we should not manipulate the `pausedSplits` here.